static/app/components/newAssigneeSelectorDropdown.tsx

```diff
@@ -0,0 +1,484 @@
+              assignableTeam => assignableTeam.team.id === owner.id
+            );
+            if (matchingTeam) {
+              return {
+                ...owner,
+                assignee: matchingTeam,
+              };
+            }
+          }
+
+          return null;
+        })
+        .filter((owner): owner is SuggestedAssignee => !!owner);
+    }
+
+    const {suggestedOwners} = state;
+    if (!suggestedOwners) {
+      return [];
+    }
+
+    const uniqueSuggestions = uniqBy(suggestedOwners, owner => owner.owner);
+    return uniqueSuggestions
+      .map<SuggestedAssignee | null>(suggestion => {
+        const [suggestionType, suggestionId] = suggestion.owner.split(':');
+        const suggestedReasonText = suggestedReasonTable[suggestion.type];
+        if (suggestionType === 'user') {
+          const member = currMembers.find(user => user.id === suggestionId);
+          if (member) {
+            return {
+              id: suggestionId,
+              type: 'user',
+              name: member.name,
+              suggestedReason: suggestion.type,
+              suggestedReasonText,
+              assignee: member,
+            };
+          }
+        } else if (suggestionType === 'team') {
+          const matchingTeam = currAssignableTeams.find(
+            assignableTeam => assignableTeam.id === suggestion.owner
+          );
+          if (matchingTeam) {
+            return {
+              id: suggestionId,
+              type: 'team',
+              name: matchingTeam.team.name,
+              suggestedReason: suggestion.type,
+              suggestedReasonText,
+              assignee: matchingTeam,
+            };
+          }
+        }
+
+        return null;
+      })
+      .filter((owner): owner is SuggestedAssignee => !!owner);
+  };
+
+  const getAssignableTeams = (): AssignableTeam[] => {
+    const teams = ProjectsStore.getBySlug(group?.project.slug)?.teams ?? [];
+    return teams
+      .sort((a, b) => a.slug.localeCompare(b.slug))
+      .map(team => ({
+        id: buildTeamId(team.id),
+        display: `#${team.slug}`,
+        email: team.id,
+        team,
+      }));
+  };
+
+  const handleSelect = async (selectedOption: SelectOption<string> | null) => {
+    // selectedOption is falsey when the option selected is currently selected
+    if (!selectedOption) {
+      setState({...state, loading: true});
+      await handleClear();
+      setState({loading: false, assignedTo: '', assignedToType: ''});
+      return;
+    }
+    // See makeMemberOption and makeTeamOption for how the value is formatted
+    const type = selectedOption.value.startsWith('USER_') ? 'user' : 'team';
+    const assigneeId =
+      type === 'user'
+        ? selectedOption.value.split('USER_')[1]
+        : selectedOption.value.split('TEAM_')[1];
+
+    if (state.assignedTo && assigneeId === state.assignedTo?.id) {
+      setState({...state, loading: true});
+      await handleClear();
```

Comment:
One thing you're missing with your requests here is error handling. As of now if anything goes wrong it will appear as if it worked.

You can use our new `useMutation` hook if you'd like to have built-in loading and error states. You can read up on that here https://develop.sentry.dev/frontend/network-requests/#mutations-postputdelete-requests

I'd suggest changing this to something like:

```tsx
const {isFetching} = useMutation<ResponseType>({
  mutationFn: (newAssignee) => {
    if (assignee.type === 'user') {
      return assignToUser({...})
    }
    ...
  },
  onSuccess: (data) => {
    // Call onAssign()
  },
  onError: () => {
    addErrorMessage('Failed to change issue assignee')
  }
})
```